product-principles/README.md

```diff
@@ -4,3 +4,8 @@ As we scale Tilt the company and the product, we will explore how to build a pro
 
 ## Usability over documentation
 We believe that Tilt should be usable without reading a separate manual. We strive to design usable interfaces, even for complex features with powerful functionality, without relying on subsequent documentation as a crutch for poor design. We prefer to use our [documentation](https://docs.tilt.dev/) to provide context on broad features, walkthroughs, and API details.
+
+## Declarative over imperative
```

Comment:
I've personally been trying to move away from the "declarative over imperative" framing. I think it's misleading in a few ways. It puts the emphasis on the programming language you use to configure the system, rather than on the mental model of how the system works.

A lot of newcomers to the Kubernetes space have interpreted the philosophy to be "let's just slap a YAML configuration file on our existing system and call it a day", which is not really what the Kubernetes people are getting at. 

It's more that we want control loops operating on infrastructure-as-data

https://twitter.com/kelseyhightower/status/1164194470436302848

and that declarative interfaces are a natural consequence of that principle, rather than the principle itself

Would it be better if I tried taking a crack at this section? I've been trying to borrow more language from the cybernetics literature (https://en.wikipedia.org/wiki/Cybernetics), which is what they're trying to invoke.

Reply:
> Would it be better if I tried taking a crack at this section?

Yes please @nicks !

Closing this PR